homeassistant/components/loqed/config_flow.py

```diff
@@ -0,0 +1,166 @@
+            api = loqed.LoqedAPI(apiclient)
+            lock = await api.async_get_lock(
+                selected_lock["backend_key"],
+                selected_lock["bridge_key"],
+                selected_lock["local_id"],
+                selected_lock["bridge_ip"],
+            )
+
+            # checking getWebooks to check the bridgeKey
+            await lock.getWebhooks()
+            return {
+                "lock_key_key": selected_lock["key_secret"],
+                "bridge_key": selected_lock["bridge_key"],
+                "lock_key_local_id": selected_lock["local_id"],
+                "bridge_mdns_hostname": selected_lock["bridge_hostname"],
+                "bridge_ip": selected_lock["bridge_ip"],
+                "name": selected_lock["name"],
+                "id": selected_lock["id"],
+            }
+        except StopIteration:
+            raise InvalidAuth from StopIteration
+        except aiohttp.ClientError:
+            _LOGGER.error("HTTP Connection error to loqed lock")
+            raise CannotConnect from aiohttp.ClientError
+
+    async def async_step_zeroconf(
+        self, discovery_info: ZeroconfServiceInfo
+    ) -> FlowResult:
+        """Handle zeroconf discovery."""
+        host = discovery_info.host
+        self._host = host
+
+        session = async_get_clientsession(self.hass)
+        apiclient = loqed.APIClient(session, f"http://{host}")
+        api = loqed.LoqedAPI(apiclient)
+        lock_data = await api.async_get_lock_details()
+
+        # Check if already exists
+        await self.async_set_unique_id(lock_data["bridge_mac_wifi"])
+        self._abort_if_unique_id_configured({CONF_HOST: host})
+
+        return await self.async_step_user()
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Show userform to user."""
+        user_data_schema = (
+            vol.Schema(
+                {
+                    vol.Required(CONF_API_TOKEN): str,
+                }
+            )
+            if self._host
+            else vol.Schema(
+                {
+                    vol.Required(CONF_NAME): str,
+                    vol.Required(CONF_API_TOKEN): str,
+                }
+            )
+        )
+
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user",
+                data_schema=user_data_schema,
+                description_placeholders={
+                    "config_url": "https://integrations.production.loqed.com/personal-access-tokens",
+                },
+            )
+
+        errors = {}
+
+        try:
+            info = await self.validate_input(self.hass, user_input)
+        except CannotConnect:
+            errors["base"] = "cannot_connect"
+        except InvalidAuth:
+            errors["base"] = "invalid_auth"
+        else:
+            await self.async_set_unique_id(
```

Comment:
Pass `raise_on_progress=False` to make the user flow always take prefence over the discovered flow.